Rollbar/AspNetCore/RollbarLogger.cs

```diff
@@ -0,0 +1,251 @@
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <param name="rollbarConfig">The rollbar configuration.</param>
+        /// <param name="rollbarOptions">The options.</param>
+        public RollbarLogger(string name, IRollbarConfig rollbarConfig, RollbarOptions rollbarOptions)
+        {
+            this._name = name;
+            this._rollbarOptions = rollbarOptions;
+
+            //perThreadInstanceCount++;
+            //if (rollbar.Config == null || string.IsNullOrWhiteSpace(rollbar.Config.AccessToken))
+            //{
+            //    rollbar.Configure(rollbarConfig);
+            //}
+            this._rollbar = RollbarFactory.CreateNew(false).Configure(rollbarConfig);
+        }
+
+        /// <summary>
+        /// Gets the name.
+        /// </summary>
+        /// <value>
+        /// The name.
+        /// </value>
+        public string Name { get { return this._name; } }
+
+        /// <summary>
+        /// Checks if the given <paramref name="logLevel" /> is enabled.
+        /// </summary>
+        /// <param name="logLevel">level to be checked.</param>
+        /// <returns>
+        ///   <c>true</c> if enabled.
+        /// </returns>
+        public bool IsEnabled(mslogging.LogLevel logLevel)
+        {
+            return _rollbarOptions.Filter(_name, logLevel);
+        }
+
+        /// <summary>
+        /// Writes a log entry.
+        /// </summary>
+        /// <typeparam name="TState"></typeparam>
+        /// <param name="logLevel">Entry will be written on this level.</param>
+        /// <param name="eventId">Id of the event.</param>
+        /// <param name="state">The entry to be written. Can be also an object.</param>
+        /// <param name="exception">The exception related to this entry.</param>
+        /// <param name="formatter">Function to create a <c>string</c> message of the <paramref name="state" /> and <paramref name="exception" />.</param>
+        public void Log<TState>(
+            mslogging.LogLevel logLevel
+            , mslogging.EventId eventId
+            , TState state
+            , Exception exception
+            , Func<TState, Exception, string> formatter
+            )
+        {
+            if (!this.IsEnabled(logLevel))
+                return;
+
+            if (state == null && exception == null)
+                return;
+
+            var currentContext = GetCurrentContext();
+
+            // let's custom build the Data object that includes the exception 
+            // along with the current HTTP request context:
+
+            string message = null;
+            if (formatter != null)
+            {
+                message = formatter(state, exception);
+            }
+
+            Rollbar.DTOs.Body payloadBody = null;
+            if (exception != null)
+                payloadBody = new DTOs.Body(exception);
+            else if (message != null)
+                payloadBody = new DTOs.Body(new DTOs.Message(message));
+
```

Comment:
It is possible for payloadBody to be null here, the case where `state != null && formatter == null` would be one such case.

Reply:
i think it is all covered by preceding checks. Line 101 makes sure that exception object is not null. The message string is derived from exception  if valid formatter is specified (Line 110). If the derived message object is valid it is used instead of the exception object to generate a  payloadBody. 
However, I now did optimize instantiation of the paloadBody a bit.